Assets/Scripts/Inventory/InventorySelector.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace FunMath
+{
+    [RequireComponent(typeof(Inventory))]
+    public class InventorySelector : MonoBehaviour
+    {
+        // Hold a reference to the inventory
+        // ASSUMPTIONS: Inventory is NEVER empty. It is filled with empty item slots.
+        [SerializeField]
+        Inventory data;
+
+        // Index of the current data
+        // Starts at -1 as there is nothing in inventory
+        [SerializeField]
+        int index = 0; //SERIALIZE IS ONLY FOR DEBUGGING
+
+        public void SwitchItemRight()
+        {
+            // Round back to 0 when at rightmost slot
+            if(index + 1 >= data.GetInventoryItemSlots().Count)

```

Comment:
If I understood the earlier part, the Inventory will always have all its slots filled, but with nothing.  So does that mean, if we `SwitchItemRight`, we'll have to go through all those empty slots before it wraps back around?  It would be nice to have a way to switch to the right and if there is nothing in the slot, we revert back to zero.  And switch left will wrap from zero to the last occupied slot.

Reply:
I guess, it can be nice for the UI to query each slot just to see what is there (though you could have a method that returns the current item type and count at all the slots).  For UI usability though, the UI will have to have logic to check the return type and keep cycling the index until the selector returns a non-null value.  I think this is fine, just wanted to point it out if you didn't expect that.
It may even make sense that this doesn't have switching methods, it just has a `SetCurrentIndex` method, and the UI is in charge of telling it what it's index is.

Reply:
I'm not a big fan of the UI informing the data what to do. It'll be trivial to add some functionality that traverses past empty item slots.